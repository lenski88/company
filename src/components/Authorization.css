.Authorization {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border: 0.2em solid  #fbfc55;
  padding: 3em;
  margin-top:25vh
}

.Authorization .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}
