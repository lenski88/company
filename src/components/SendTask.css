.SendTask {
    display:flex;
    background-color: #0301a8 ;
    flex-flow: column wrap;
    align-items: center;
    position:absolute;
    top: 20%;
    width:100%;
    z-index: 1;
}

.SendTask .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
    box-shadow:  0.4em 0.4em  #b2b2b8e8;
}

@media (max-width: 1033px) {
    .SendTask .inputText {width: 90%;}
  }

 
