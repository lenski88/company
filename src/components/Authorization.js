import React from "react";
import PropTypes from "prop-types";

import "./Authorization.css";

class Authorization extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired /* массив сотрудников */,
    login:
      PropTypes.bool.isRequired /* прошла ли авторизация; по умолчанию false */,
    cbGetUser: PropTypes.func,
  };

  state = {
    login: this.props.login,
    nameUser: "",
    errName: "",
    button: false,
  };

  nameUser = (eo) => {
    this.setState({
      nameUser: eo.target.value,
    });

    let arrayNames = this.props.employes;
    let name = eo.target.value;
    let isLogin = arrayNames.some((i) => {
      return name === i.login;
    });

    if (!isLogin) {
      this.setState({
        errName: "Нет такого сотрудника",
      });
    } else {
      this.setState({
        errName: "",
      });
    }
  };

  getUser = () => {
    if (!this.state.errName && !this.state.errPassword) {
      let arrayNames = this.props.employes;
      let name = this.state.nameUser;
      let user = arrayNames.find((i) => {
        return name === i.login;
      });
      this.props.cbGetUser(user);
      this.setState({
        nameUser: "",
        login: this.props.login,
      });
    }
  };

  render() {
    return (
      !this.state.login && (
        <div className={!this.props.login?"Authorization": "NoAuthorization"}>
          <label>Введите логин:</label>
          <br />
          <input
            className="inputText"
            type="text"
            name="nameUser"
            value={this.state.nameUser}
            onChange={this.nameUser}
            autoFocus
          ></input>
          <br />
          <span>{this.state.errName}</span>
          <br />
          <br />
          {this.state.nameUser && (
            <input
              className="button"
              type="button"
              onPointerDown={this.getUser}
              value="[Войти]"
              disabled={this.state.button}
            ></input>
          )}
        </div>
      )
    );
  }
}

export default Authorization;
