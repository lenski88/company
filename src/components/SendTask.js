import React from "react";
import PropTypes from "prop-types";

import "./SendTask.css";

class SendTask extends React.Component {
  static propTypes = {
    employes: PropTypes.array,
    idEmp: PropTypes.number, ////id сотрудника, которому будет поставлена задача
    cbNewTask: PropTypes.func,
    cbExitSendTask:PropTypes.func
  };

  state = {
    newTask: "",
    emptyTask: true,
  };

  empName = () => {
    let employes = this.props.employes;
    let idEmp = this.props.idEmp;
    let emp = employes.find((i) => {
      return i.id === idEmp;
    });
    let name = emp.name.split(" ").splice(0, 2).join(" ");
    return name;
  };

  setNewTask = (eo) => {
    if (eo.target.value.trim() === "") {
      this.setState({
        emptyTask: true,
      });
    } else {
      this.setState({
        emptyTask: false,
      });
    }
    this.setState({
      newTask: eo.target.value,
    });
  };

  sendTask = () => {
    let empIndex = this.props.employes.findIndex((i) => {
      return i.id === this.props.idEmp;
    });
    let arrayTask = this.props.employes[empIndex].task; //массив задач сотрудника
    let lastIdTask; //последний id в массиве задач

    arrayTask.length === 0? lastIdTask = 0:lastIdTask = arrayTask[arrayTask.length - 1].id; // если массив задач пустой
    
    let sender = this.props.user.name;
    arrayTask = [
      ...arrayTask,
      { id: lastIdTask + 1, task: this.state.newTask, sender: sender },
    ];

    let emp = this.props.employes[empIndex]; // сотрудник с обновленным списком задач
    emp = { ...emp, task: arrayTask };

    this.props.cbNewTask(emp);

    this.setState({
      newTask: "",
    });
  };

  exitSendTask = () => {
    let isExit = window.confirm("Изменения не сохранятся! Выйти?")
    if(isExit) {
      this.setState({
        newTask:'',
        emptyTask:true
      })
      this.props.cbExitSendTask();
    } else {
      return;
    }
  }

  render() {
    return (
      <div className="SendTask">
        <div style={{ color: "#fbfc55", textAlign: "center" }}>
          <h2>Новая задача</h2>
          <h4 id='empName'>Кому:{this.empName()} </h4>
        </div>
        <textarea
          className="inputText"
          maxLength="400"
          cols="40"
          rows="10"
          placeholder="Поле для ввода..."
          value={this.state.newTask}
          onChange={this.setNewTask}
        ></textarea>
        {!this.state.emptyTask && (
          <input
            className="button"
            type="button"
            value="[Отправить]"
            onPointerDown={this.sendTask}
          ></input>
        )}
        <input className="button" type="button" value="[Отмена]" onPointerDown={this.exitSendTask}></input>
      </div>
    );
  }
}

export default SendTask;
