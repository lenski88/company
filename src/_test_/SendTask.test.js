"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { employes } from "./Authorization.test";

import SendTask from "../components/SendTask";

test("рендерится ли кнопка 'отмена'", () => {
  let component = renderer.create(
    <SendTask employes={employes} idEmp={employes[4].id} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let button = component.root.find((el) => el.props.value == "[Отмена]");
  expect(button).toBeTruthy();
});

test("правильность отображения имени адресата", () => {
  let component = renderer.create(
    <SendTask employes={employes} idEmp={employes[4].id} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let empName = component.root.find((el) => el.props.id == "empName");
  expect(empName.children.join("").trim()).toBe("Кому:Кулаков Кондрат");
});

test("рендерится ли кнопка 'отправить' при заполненном поле ввода", () => {
  let component = renderer.create(
    <SendTask employes={employes} idEmp={employes[4].id} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let input = component.root.find((el) => el.props.className == "inputText");
  input.props.onChange({ target: { value: "dk;gasjkdghlkjahdlkjahldkj" } });

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let buttonSend = component.root.find((el) => el.props.value == "[Отправить]");
  expect(buttonSend).toBeTruthy();
});
