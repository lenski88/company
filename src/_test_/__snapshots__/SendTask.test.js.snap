// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`правильность отображения имени адресата 1`] = `
<div
  className="SendTask"
>
  <div
    style={
      Object {
        "color": "#fbfc55",
        "textAlign": "center",
      }
    }
  >
    <h2>
      Новая задача
    </h2>
    <h4
      id="empName"
    >
      Кому:
      Кулаков Кондрат
       
    </h4>
  </div>
  <textarea
    className="inputText"
    cols="40"
    maxLength="400"
    onChange={[Function]}
    placeholder="Поле для ввода..."
    rows="10"
    value=""
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Отмена]"
  />
</div>
`;

exports[`проверка Authorization 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка Authorization 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="sdfasdf"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="saveliev47"
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка нажатия на кнопку войти 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка рендера кнопки при верном логине 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка рендера кнопки при верном логине 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`рендерится ли кнопка 'отмена' 1`] = `
<div
  className="SendTask"
>
  <div
    style={
      Object {
        "color": "#fbfc55",
        "textAlign": "center",
      }
    }
  >
    <h2>
      Новая задача
    </h2>
    <h4
      id="empName"
    >
      Кому:
      Кулаков Кондрат
       
    </h4>
  </div>
  <textarea
    className="inputText"
    cols="40"
    maxLength="400"
    onChange={[Function]}
    placeholder="Поле для ввода..."
    rows="10"
    value=""
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Отмена]"
  />
</div>
`;

exports[`рендерится ли кнопка 'отправить' при заполненном поле ввода 1`] = `
<div
  className="SendTask"
>
  <div
    style={
      Object {
        "color": "#fbfc55",
        "textAlign": "center",
      }
    }
  >
    <h2>
      Новая задача
    </h2>
    <h4
      id="empName"
    >
      Кому:
      Кулаков Кондрат
       
    </h4>
  </div>
  <textarea
    className="inputText"
    cols="40"
    maxLength="400"
    onChange={[Function]}
    placeholder="Поле для ввода..."
    rows="10"
    value=""
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Отмена]"
  />
</div>
`;

exports[`рендерится ли кнопка 'отправить' при заполненном поле ввода 2`] = `
<div
  className="SendTask"
>
  <div
    style={
      Object {
        "color": "#fbfc55",
        "textAlign": "center",
      }
    }
  >
    <h2>
      Новая задача
    </h2>
    <h4
      id="empName"
    >
      Кому:
      Кулаков Кондрат
       
    </h4>
  </div>
  <textarea
    className="inputText"
    cols="40"
    maxLength="400"
    onChange={[Function]}
    placeholder="Поле для ввода..."
    rows="10"
    value="dk;gasjkdghlkjahdlkjahldkj"
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Отправить]"
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Отмена]"
  />
</div>
`;
